OrbitQt/orbitmainwindow.cpp

```diff
@@ -631,15 +644,61 @@ void OrbitMainWindow::OpenCapture(const std::string& filepath) {
   ui->MainTabWidget->setCurrentWidget(ui->CaptureTab);
 }
 
-void OrbitMainWindow::OpenDisassembly(std::string a_String, DisassemblyReport report) {
-  auto* dialog = new OrbitDisassemblyDialog(this);
-  dialog->SetText(std::move(a_String));
-  dialog->SetDisassemblyReport(std::move(report));
-  dialog->setWindowTitle("Orbit Disassembly");
-  dialog->setAttribute(Qt::WA_DeleteOnClose);
-  dialog->setWindowFlags(dialog->windowFlags() | Qt::WindowMinimizeButtonHint |
-                         Qt::WindowMaximizeButtonHint);
-  dialog->show();
+static std::pair<QString, QString> SplitLineAtFirstLinebreak(const std::string& input) {
+  const auto end_of_first_line = input.find('\n');
+  if (end_of_first_line == std::string::npos) {
+    return std::make_pair(QString{}, QString::fromStdString(input));
+  }
+
+  const auto first = std::string_view{input}.substr(0, end_of_first_line);
+  const auto other = std::string_view{input}.substr(end_of_first_line + 1);
+  return std::make_pair(QString::fromUtf8(first.data(), first.size()),
+                        QString::fromUtf8(other.data(), other.size()));
+}
+
+void OrbitMainWindow::OpenDisassembly(std::string src, DisassemblyReport report) {
+  auto [title, code] = SplitLineAtFirstLinebreak(src);
+
+  // Since we removed the first line to be used as a title, line numbers are shifted between
+  // code and report.
+  auto line_shift = 1;
+
+  if (code.startsWith("Platform:")) {
+    const auto pos_newline = code.indexOf('\n');
+    if (pos_newline != -1) {
+      code = code.mid(pos_newline + 1);
+      line_shift++;
+    }
+  }
+
+  const auto number_of_code_lines = code.count('\n');
+  QVector<int> hit_counts;
+  hit_counts.reserve(number_of_code_lines);
+
+  QVector<double> hit_ratios;
+  hit_ratios.reserve(number_of_code_lines);
+
+  QVector<double> total_hit_ratios;
+  total_hit_ratios.reserve(number_of_code_lines);
+
+  const auto total_number_of_samples = report.GetNumSamples();
+  const auto number_of_samples_in_function = report.GetNumSamplesInFunction();
+
+  for (int i = 0; i < number_of_code_lines; i++) {
```

Comment:
As said above, I would rather store the disassembly report into the Codeview (I think there is a super-class).

Reply:
Resolving this for now since this radically changed.